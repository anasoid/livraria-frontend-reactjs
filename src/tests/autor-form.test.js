import React from 'react';
import renderer from 'react-test-renderer';
import AutorForm from '../components/autor/autor-form';
import {mount} from 'enzyme/build/index';

describe('test autorForm component', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = mount(<AutorForm />);
  });

  it('should renders without crashing', () => {
    const tree = renderer.create(<AutorForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('AutorConteiner should be contain form', () => {
    expect(wrapper.find('form').at(0)).toExist();
  });

  it('should be contains nome input', () => {
    const nome = wrapper.find('#autorNome').at(0);
    expect(nome).toMatchSelector('input');
    expect(nome).toHaveProp('name', 'nome');
  });

  it('should be contains email input', () => {
    const email = wrapper.find('#autorEmail').at(0);
    expect(email).toMatchSelector('input');
    expect(email).toHaveProp('name', 'email');
  });

  it('should be contain a button "Gravar Autor" and type submit', () => {
    const button = wrapper.find('#btnGravarAutor').at(0);
    expect(button).toHaveProp('type', 'submit');
    expect(button).toHaveText('Gravar Autor');
  });

});