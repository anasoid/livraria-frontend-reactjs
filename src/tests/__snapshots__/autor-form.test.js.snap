// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test autorForm component should renders without crashing 1`] = `
<div>
  <form>
    <label
      htmlFor="autorNome"
    >
      Nome: 
    </label>
    <input
      id="autorNome"
      name="nome"
    />
    <br />
    <label
      htmlFor="autorEmail"
    >
      Email: 
    </label>
    <input
      id="autorEmail"
      name="email"
    />
    <br />
    <button
      id="btnGravarAutor"
      type="submit"
    >
      Gravar Autor
    </button>
  </form>
</div>
`;
